File: 2441_largest_positive_integer_that_exists_with_its_negative/findMaxK.py
class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        nums.sort()

        l, r = 0, len(nums) - 1

        while l < r:
            if nums[l] < 0 and abs(nums[l]) == nums[r]:
                return nums[r]
            if abs(nums[l]) > nums[r]:
                l += 1
            else:
                r -= 1

        return -1

File: 2441_largest_positive_integer_that_exists_with_its_negative/findMaxK.java
class Solution {
    public int findMaxK(int[] nums) {
        Arrays.sort(nums);

        int left = 0;
        int right = nums.length - 1;

        while (left < right) {
            if (nums[left] + nums[right] == 0) {
                return nums[right];
            } else if (nums[left] + nums[right] < 0) {
                left++;
            } else {
                right--;
            }
        }

        return -1;
    }
}